### PR TITLE
Check if txs and mbs exists in es

### PR DESCRIPTION
**Pull request by miiu96**
### Bug fix
`Bug`: Status of transactions that are indexed in elasticserach database remains Pending because cross shard transactions was indexed first by the observer from destination shard and after that was indexed again by  the observer from source shard. This happens because observer from source shard was behind. 
`Fix` : Before index a transaction an extra check was added to verify if transaction exists in database.

This bug also occurs when mini-blocks are indexed

### CODE REVIEW

File: core/indexer/common.go [hunk 1]
```diff
@@ -239,13 +239,29 @@ func canInterpretAsString(bytes []byte) bool {
 	return true
 }
 
-func serializeBulkMiniBlocks(hdrShardID uint32, bulkMbs []*Miniblock) bytes.Buffer {
+func serializeBulkMiniBlocks(
+	hdrShardID uint32,
+	bulkMbs []*Miniblock,
+	foundedObjMap func(hashes []string, index string) (map[string]bool, error),
```
Comment (ccorcoveanu) [inline core/indexer/common.go:245]: Should rename to something like `getAlreadyIndexedItems`. Just an idea, but `foundedObjMap` does not help to understand what it does

File: core/indexer/common.go [hunk 2]
```diff
@@ -254,12 +270,18 @@ func serializeBulkMiniBlocks(hdrShardID uint32, bulkMbs []*Miniblock) bytes.Buff
...
 			}
-
 		} else {
 			// update miniblock
 			meta = []byte(fmt.Sprintf(`{ "update" : { "_id" : "%s", "_type" : "%s"  } }%s`, mb.Hash, "_doc", "\n"))
-			serializedData = []byte(fmt.Sprintf(`{ "doc" : { "receiverBlockHash" : "%s" } }`, mb.ReceiverBlockHash))
+			if hdrShardID == mb.SenderShardID {
+				// update sender block hash
+				serializedData = []byte(fmt.Sprintf(`{ "doc" : { "senderBlockHash" : "%s" } }`, mb.SenderBlockHash))
+			} else {
```
Comment (ccorcoveanu) [inline core/indexer/common.go:279]: Can we remove the else here so we'd have less indentation?
```go
serializedData = []byte(fmt.Sprintf(`{ "doc" : { "receiverBlockHash" : "%s" } }`, mb.ReceiverBlockHash))
if hdrShardID == mb.SenderShardID {
    // update sender block hash
    serializedData = []byte(fmt.Sprintf(`{ "doc" : { "senderBlockHash" : "%s" } }`, mb.SenderBlockHash))
}
```

File: core/indexer/common.go [hunk 3]
```diff
@@ -282,18 +304,35 @@ func prepareBufferMiniblocks(buff bytes.Buffer, meta, serializedData []byte) byt
...
+		log.Warn("indexer", "error", err.Error())
+		return buff
+	}
+
 	for _, tx := range bulk {
 		var meta, serializedData []byte
-
-		if isCrossShardDstMe(tx, selfShardID) && tx.Status != txStatusInvalid {
-			// update tx
+		if existsInDb[tx.Hash] {
```
Comment (ccorcoveanu) [inline core/indexer/common.go:328]: maybe extract this if/else into a function like `getSerializedData`?

File: core/indexer/common.go (hunk 1 above)
Comment (miiu96) [reply core/indexer/common.go:245]: good suggestion. renamed.

File: core/indexer/common.go (hunk 2 above)
Comment (miiu96) [reply core/indexer/common.go:279]: maybe we can remove the `else` from this `if` but if we do that we will have two assignments in variable `serializedData`  and one of them is pointless

File: core/indexer/common.go (hunk 3 above)
Comment (miiu96) [reply core/indexer/common.go:328]: done.

**Pull request merged**